delayScanPlots.py

```diff
@@ -1,48 +1,81 @@
-
+import sys
 import matplotlib.colors as mcolors
+import pandas as pd
+import mplhep as hep
+plt.style.use(hep.style.CMS)
+plt.rcParams.update({'font.size': 18})
+import json
+import glob
+import pandas as pd
+import mplhep as hep
+plt.style.use(hep.style.CMS)
+plt.rcParams.update({'font.size': 18})
+import json
+import glob
 
-import mplhep
-mplhep.style.use(mplhep.style.CMS)
-
-
-def delay_scan_plots(fname,
-                     title='ETx Delay Scan',
-                     outputFileName=None,
-                     ECOND=False):
-
-    data = np.load(f'./{fname}.npz',allow_pickle=True )
-    x = data['errorcounts'].flatten()[0]
-    y = data['bitcounts'].flatten()[0]
-    errorRates = []
-
-    for i in range(6):
-       errorRates.append(list(np.array(x[i])/np.array(y[i])))
-
-    errorRates = np.array(errorRates).T.flatten()
-
-    if ECOND:
-        a,b=np.meshgrid(np.arange(6),np.arange(63))
-    else:
-        a,b=np.meshgrid(np.arange(13),np.arange(63))
 
-    fig,ax=plt.subplots()
+def jsonload(fname):
+    with open(fname) as jsonfile:
+        try:
+            return json.load(jsonfile)
+        except Exception:
+            print(fname)
 
-    plt.hist2d(a.flatten(),b.flatten(),
-               weights=errorRates,
-               bins=(np.arange(7)-0.5,np.arange(64)-0.5),
-               cmap='RdYlBu_r',
-               alpha=errorRates>0,
-               figure=fig);
 
-    plt.colorbar().set_label(label='Transmission errors rate ',size=32)
-    plt.title(title,size=32)
-    plt.ylabel('Phase Select Setting', size=32)
-    plt.xlabel('Channel Number', size=32)
-    if outputFileName:
-        plt.savefig(f'./ECOND_delay_scan_board{boardNum}_{freq}.png',dpi=300, facecolor = "w")
-    plt.close(fig)
+def json_delayscan(indir = ""):
+    fdir = f"{indir}"
+    fnames = list(np.sort(glob.glob(f"{fdir}/*.json")))
+    fnames1 = np.array([y.split(".") for y in [x for x in np.array([x.split("/") for x in fnames])[:,-1]]])[:,0]
+    data = [jsonload(fname) for fname in fnames]
+    etx_delays = []
+    etx_bitcounts = []
+    etx_errcounts = []
+    for i in range(len(data)):
+        for j in range(len(data[i]['tests'])):
+            if 'metadata' in data[i]['tests'][j]:
+                if "test_io.py::test_eTX_delayscan" in data[i]['tests'][j]['nodeid']:
+                    etx_delays.append(data[i]['tests'][j]['metadata']['eTX_delays'])
+                    etx_bitcounts.append(data[i]['tests'][j]['metadata']['eTX_bitcounts'])
+                    etx_errcounts.append(data[i]['tests'][j]['metadata']['eTX_errcounts'])
+    etx_delays = np.array(etx_delays)
+    etx_bitcounts = np.array(etx_bitcounts)
+    etx_errcounts = np.array(etx_errcounts)
+    etx_rate = etx_errcounts / etx_bitcounts
+    return fnames1, etx_rate
 
-    return fig
+def delay_scan_plots(fname=None,indir=None,boardNum=None,freq=None, ECOND=False):
+    if ECOND == True:
+        boardNum, new_array = json_delayscan(indir)
+        for i in range(len(boardNum)):
```

Comment:
We want this function to return a single plot, not save many plots.  Instead of passing in a directory, either pass in a single filename or single array, and move the loop to the script that is calling it